### PR TITLE
Added missing tests for the go/streamlog package

### PR DESCRIPTION
**Pull request by VaibhavMalik4187**
## Description
This commit increased the code coverage of the `go/streamlog` package to 93%.

## Related Issue(s)
Partially addresses: https://github.com/vitessio/vitess/issues/14931

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/streamlog/streamlog_test.go [hunk 1]
```diff
@@ -260,3 +264,152 @@ func TestFile(t *testing.T) {
 		t.Errorf("streamlog file: want %q got %q", want, got)
 	}
 }
+
+func TestSetAndGetMethods(t *testing.T) {
+	SetRedactDebugUIQueries(false)
```
Comment (GuptaManan100) [inline go/streamlog/streamlog_test.go:269]: Whenever we change the global state in a test, we should ensure we set it back in a defer function call.

So, we should add this code to the test -

```go
originalVal := GetRedactDebugUIQueries()
defer func() {
    SetRedactDebugUIQueries(originalVal)
}()
```

We should do this for all the tests that change any state.

**Review by GuptaManan100 — CHANGES_REQUESTED** [review]
The following change has to be made to the tests.

**Review by GuptaManan100 — COMMENTED** [review]
`TestRegisterStreamLogFlags` and `TestSetAndGetMethods` don't seem very useful. Please remove those two tests, and we can merge the remaining two.

**Comment by VaibhavMalik4187** [comment]
> `TestRegisterStreamLogFlags` and `TestSetAndGetMethods` don't seem very useful. Please remove those two tests, and we can merge the remaining two.

Alright, I'll update the code soon.

**Comment by VaibhavMalik4187** [comment]
> `TestSetAndGetMethods`

Done

**Review by GuptaManan100 — APPROVED** [review]

**Review by harshit-gangal — APPROVED** [review]

**Pull request merged**